tests/test_lockable_component.cairo

```diff
@@ -0,0 +1,178 @@
+        .deploy(
+            @array![883045738439352841478194533192765345509759306772397516907181243450667673002]
+        )
+        .unwrap();
+
+    // mint a new token
+    let dispatcher = IERC721Dispatcher { contract_address: erc721_contract_address };
+    dispatcher.mint(recipient, 1.try_into().unwrap());
+
+    // deploy account contract
+    let account_contract = declare("AccountPreset").unwrap();
+    let mut acct_constructor_calldata = array![erc721_contract_address.try_into().unwrap(), 1, 0];
+    let (account_contract_address, _) = account_contract
+        .deploy(@acct_constructor_calldata)
+        .unwrap();
+
+    (account_contract_address, erc721_contract_address)
+}
+
+
+#[test]
+fn test_lockable_owner() {
+    let (contract_address, erc721_contract_address) = __setup__();
+    let acct_dispatcher = IAccountDispatcher { contract_address: contract_address };
+
+    let token_dispatcher = IERC721Dispatcher { contract_address: erc721_contract_address };
+    let owner = acct_dispatcher.owner();
+    let token_owner = token_dispatcher.ownerOf(1.try_into().unwrap());
+
+    start_cheat_caller_address(contract_address, token_owner);
+
+    assert(owner == token_owner, 'invalid owner');
+    stop_cheat_caller_address(contract_address);
+}
+#[test]
+fn test_lockable() {
+    let (contract_address, _) = __setup__();
+    let acct_dispatcher = IAccountDispatcher { contract_address: contract_address };
+
+    let owner = acct_dispatcher.owner();
+
+    start_cheat_caller_address(contract_address, owner);
+
+    let lockable_dispatcher = ILockableDispatcher { contract_address };
+
+    lockable_dispatcher.lock(40);
+    let check_lock = lockable_dispatcher.is_lock();
+
+    assert(check_lock == true, 'Account Not Lock');
+    stop_cheat_caller_address(contract_address);
+}
+
+#[test]
+#[should_panic(expected: ('Account locked',))]
+fn test_should_fail_when_locked() {
+    let (contract_address, _) = __setup__();
+    let acct_dispatcher = IAccountDispatcher { contract_address: contract_address };
+    let safe_dispatcher = IExecutableDispatcher { contract_address };
+
+    let owner = acct_dispatcher.owner();
+    let lock_duration = 30_u64;
+
+    let lockable_dispatcher = ILockableDispatcher { contract_address };
+
+    start_cheat_caller_address(contract_address, owner);
+    lockable_dispatcher.lock(lock_duration);
+    stop_cheat_caller_address(contract_address);
+
+    // deploy `HelloStarknet` contract for testing
+    let test_contract = declare("HelloStarknet").unwrap();
+    let (test_address, _) = test_contract.deploy(@array![]).unwrap();
+
+    // craft calldata for call array
+    let mut calldata = array![100].span();
+    let call = Call {
+        to: test_address,
+        selector: 1530486729947006463063166157847785599120665941190480211966374137237989315360,
+        calldata: calldata
+    };
+
+    start_cheat_caller_address(contract_address, owner);
+    safe_dispatcher.execute(array![call]);
+}
+
```

Comment:
add test for `test_upgrade_should_fail_when_locked`, `test_locking_should_fail_if_already_locked` and `test_locking_should_fail_if_not_owner`